src/dbnode/storage/bootstrap/bootstrapper/uninitialized/options.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package uninitialized
+
+import (
+	"errors"
+
+	"github.com/m3db/m3/src/dbnode/storage/bootstrap/result"
+	"github.com/m3db/m3x/instrument"
+)
+
+var (
+	errInstrumentOptsNotSet = errors.New("instrument options not set")
+	errResultOptsNotSet     = errors.New("result options not set")
+)
+
+type options struct {
+	resultOpts result.Options
+	iOpts      instrument.Options
+}
+
+// NewOptions creates a new Options.
+func NewOptions() Options {
+	return &options{}
```

Comment:
hm why not initialise these to `result.NewOptions()` and `instrument.NewOptions()`?

asking because almost always `NewOptions()` returns something that's valid by default

Reply:
To avoid issues like we've had in the past where we cant find metrics because (somewhere) in the deep configuration pipeline / setup we forgot to set InstrumentOptions()

This seems like a reasonable compromise to make sure that we don't run into that type of stuff anymore.

Reply:
Don't necessarily disagree but doing it in one place like this breaks from the convention everywhere else in the code.